src/components/radio/radio_button.ts

```diff
@@ -0,0 +1,270 @@
+    this.disabled_ = (value != null && value !== false) ? true : null;
+  }
+
+  @Input('value')
+  get value() {
+    return this.value_;
+  }
+
+  set value(value: string) {
+    if (this.value_ != value) {
+      ObservableWrapper.callEmit(this.change, null);
+    }
+    this.value_ = value;
+  }
+
+  @Input('selected')
+  get selected() {
+    return this.selected_;
+  }
+
+  set selected(selected: MdRadioButton) {
+    this.selected_ = selected;
+    this.value = selected.value;
+
+    selected.checked = true;
+  }
+}
+
+
+@Component({
+  selector: 'md-radio-button',
+  templateUrl: './components/radio/radio_button.html',
+  styleUrls: ['./components/radio/radio-button.css'],
+  encapsulation: ViewEncapsulation.None
+})
+export class MdRadioButton {
+  /**
+   * Input field wrapped reference.
+   * We use a hidden native input field to handle changes to focus state via keyboard navigation,
+   * with visual rendering done separately. The native element is kept in sync with the overall
+   * state of the component.
+   */
+  @ViewChild('input') private inputRef_: ElementRef;
+
+  /** Stores the raw input element. */
+  private inputElement_: HTMLInputElement;
+
+  @HostBinding('class.md-radio-focused') private isFocused_: boolean;
+
+  /** Whether this radio is checked. */
+  checked_: boolean;
+
+  /** The unique ID for the radio button. */
+  @HostBinding('id')
+  @Input()
+  id: string;
+
+  /** Analog to HTML 'name' attribute used to group radios for unique selection. */
+  @Input() name: string;
+
+  /** Whether this radio is disabled. */
+  disabled_: boolean;
+
+  /** Value assigned to this radio.*/
+  value_: string;
+
+  /** The parent radio group. May or may not be present. */
+  radioGroup: MdRadioGroup;
+
+  /** Dispatcher for coordinating radio unique-selection by name. */
+  radioDispatcher: MdRadioDispatcher;
+
+  /** Event emitted when the group value changes. */
+  @Output() change: EventEmitter<any>;
+
+  constructor(@Optional() radioGroup: MdRadioGroup, @Attribute('id') id: string,
+              radioDispatcher: MdRadioDispatcher) {
+    // Assertions. Ideally these should be stripped out by the compiler.
+    // TODO(jelbourn): Assert that there's no name binding AND a parent radio group.
+
+    this.radioGroup = radioGroup;
+    this.radioDispatcher = radioDispatcher;
+    this.value_ = null;
+    this.checked_ = false;
+    this.change = new EventEmitter();
+    this.name = null;
+
+    this.id = id != null ? id : `md-radio-${_uniqueIdCounter++}`;
+
+    radioDispatcher.listen((name: string) => {
+      if (name == this.name) {
+        this.checked = false;
+      }
+    });
+  }
+
+  private onFocus() {
+    this.isFocused_ = true;
+  }
+
+  private onBlur() {
+    this.isFocused_ = false;
+  }
+
+  /** Input change handler, called only on keyboard selection. */
+  private onChange(ev: Event) {
+    this.checked = (<HTMLInputElement>ev.target).checked;
```

Comment:
Since radio buttons only call `change` when they're selected, could you not just say
```ts
this.checked = true;
```
?